opentimelineio_contrib/adapters/media_file.py

```diff
@@ -0,0 +1,202 @@
+#
+# Copyright 2018 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+__doc__ = """ Media file read-only adapter. This adapter just wraps a video
+or audio file in a single-clip timeline with the correct duration. """
+
+
+import os
+import subprocess
+import json
+
+import opentimelineio as otio
+
+
+def get_metadata(filepath):
+    proc = subprocess.Popen(
+        [
+            'ffprobe',
+            '-v', 'quiet',
+            '-print_format', 'json',
+            '-show_format',
+            '-show_streams',
+            filepath
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        env=os.environ
+    )
+    out, err = proc.communicate()
+    metadata = {}
+
+    metadata = json.loads(out)
```

Comment:
Looks like because of this line, you can delete line 52

Reply:
Also, might want to catch `RuntimeError` or `ValueError` in case the the json parser fails (and map them into `OTIOError`).

Reply:
...Then you could include the `err` stderr output in the exception error message.